tests/posix_semaphore/main.c

```diff
@@ -260,13 +266,20 @@ void test4(void)
             puts("first: timed out");
         }
     }
-    stop = xtimer_now_usec64() - start;
-    if ((stop < (exp - 100)) || (stop > (exp + 100))) {
-        fmt_u64_dec(uint64_str, stop);
+
+    fmt_u64_dec(uint64_str, elapsed);
```

Comment:
fmt doesn't null terminate. either you save the length and supply it to printf, or you do sth like:
```uint64_str[fmt_u64_dec(uint64_str, elapsed)] = '\0';```

Reply:
The problem was also there before. But I will fix it.